android/java/org/chromium/chrome/browser/BraveApplication.java

```diff
@@ -0,0 +1,15 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser;
+
+import android.app.Application;
+
+import org.chromium.chrome.browser.BraveSyncWorker;
+
+public class BraveApplication extends Application {
+    // Sync worker
+    public BraveSyncWorker mBraveSyncWorker;
```

Comment:
this class will never be used in brave-core so we should just leave it out

Reply:
Removed the `ChromeApplication` patch and replaced `BraveSyncWorker` with `BraveSyncService` as recommended. `BraveSyncService` doesn't yet implement the actual sync service API as the goal here was mainly to get the UI up and running with the understanding that the actual service will be implemented later.

I also fixed the preferences in `BraveSyncScreensPreference` to refer to the appropriate brave-core preferences.